Index: ../../../fltk-1.3/src/fl_images_core.cxx
===================================================================
diff --git a/branches/branch-1.3/src/fl_images_core.cxx b/branches/branch-1.3/src/fl_images_core.cxx
--- a/branches/branch-1.3/src/fl_images_core.cxx	(Revision 12079)
+++ b/branches/branch-1.3/src/fl_images_core.cxx	(Arbeitskopie)
@@ -28,6 +28,7 @@
 #include <FL/Fl_Shared_Image.H>
 #include <FL/Fl_BMP_Image.H>
 #include <FL/Fl_GIF_Image.H>
+#include <FL/Fl_Anim_GIF_Image.H>
 #include <FL/Fl_JPEG_Image.H>
 #include <FL/Fl_PNG_Image.H>
 #include <FL/Fl_PNM_Image.H>
@@ -66,7 +67,8 @@
 		int) {				// I - Amount of data (not used)
   if (memcmp(header, "GIF87a", 6) == 0 ||
       memcmp(header, "GIF89a", 6) == 0)	// GIF file
-    return new Fl_GIF_Image(name);
+    return Fl_GIF_Image::animate ? new Fl_Anim_GIF_Image(name, 0, Fl_Anim_GIF_Image::Start) :
+                                   new Fl_GIF_Image(name);
 
   if (memcmp(header, "BM", 2) == 0)	// BMP file
     return new Fl_BMP_Image(name);
